set-version.sh

```diff
@@ -1,24 +1,32 @@
 #!/bin/bash -e
 
 NEW_VERSION=$1
+NEW_NPM_VERSION=${2:-${1}}
 
-# Maven
-mvn versions:set -DnewVersion=$NEW_VERSION -DgenerateBackupPoms=false -DgroupId=org.keycloak* -DartifactId=*
+if [ $NEW_VERSION == 'use_current' ] ; then
+  # obtain NEW_VERSION from maven pom file
+  # - useful for downstream projects which already set the version in pom files
+  NEW_VERSION=$(./get-version.sh)
+  NEW_NPM_VERSION=$(echo $NEW_VERSION | awk -F '.' '{ print $1"."$2"."$3"+"$4 }')
+else
```

Comment:
What's the purpose of this?

Reply:
Follow up from https://github.com/keycloak/keycloak/pull/23425#discussion_r1334351701.
This will assign NPM compatible version string. The third dot is changed to '+' as suggested by @jonkoops .